keras_cv/layers/object_detection/nms_prediction_decoder.py

```diff
@@ -57,32 +57,35 @@ def __init__(
         self.max_detections_per_class = max_detections_per_class
         self.built = True
 
-    def call(self, box_pred, confidence_pred):
+    def call(self, box_prediction, confidence_prediction):
         """Accepts images and raw predictions, and returns bounding box predictions.
 
         Args:
-            images: Tensor of shape [batch, height, width, channels].
-            box_pred: Dense Tensor of shape [batch, boxes, 4] in the
+            box_prediction: Dense Tensor of shape [batch, boxes, 4] in the
                 `bounding_box_format` specified in the constructor.
-            confidence_pred: Dense Tensor of shape [batch, boxes, num_classes].
+            confidence_prediction: Dense Tensor of shape [batch, boxes, num_classes].
         """
-        box_pred = bounding_box.convert_format(
-            box_pred,
+        target_format = bounding_box.preserve_rel(
```

Comment:
Doesn't this require `image_shape` or `image` later during the conversion? I'm not sure why we prefer converting to relative format

Reply:
preserve_rel makes it so you don't need image shape.

it matches the input's relative/nonrelative status, so you can use it for operations where relative/nonrelative doesn't matter.

Reply:
(IoU comptuation/NMS falls in that category)

Reply:
But how does the box gets converted to a relative format without an image shape?
Also what's the benefit here? Given this layer is running on CPU today, I'd rather minimize the ops we need to call

Reply:
it doesn't.  There's no extra ops.  preserve_rel() matches the relative/nonrelative format of the source format in the target format.

It is still a single conversion - check out the code of preserve_rel().  What it does is make it so if you're in yxyx you stay in yxyx, but if you're in rel_yxyx you stay in rel_yxyx.

This makes it so you don't need images to apply NMS, but your outputs are the same.   Its strictly better - and theres no performance cost.

Reply:
(preserve_rel is purely a python space construct.  It returns a format name based on the target format to prevent calls that require image shapes)

Reply:
Oh....that's a really confusing name then....